src/box/lua/xlog.c

```diff
@@ -75,20 +75,34 @@ lbox_checkcursor(struct lua_State *L, int narg, const char *src)
 
 /* {{{ Xlog Parser */
 
+/**
+ * Replaces whitespace with underscore for xlog key names, e.g.
+ * "page index offset" => "page_index_offset".
+ */
+static void
+lbox_xlog_pushkey(lua_State *L, const char *key)
+{
+	luaL_Buffer b;
+	luaL_buffinit(L, &b);
+	for (const char *pos = key; *pos; pos++)
+		luaL_addchar(&b, (*pos != ' ') ? *pos : '_');
+	luaL_pushresult(&b);
+}
+
```

Comment:
I asked the same question in Tarantool Dev and @kostja said what whitespaces are better for error messages. As result, I decided to keep whitespaces in iproto_key_strs for error messages and use underscores in Lua.